selector-set.js

```diff
@@ -1,412 +1,410 @@
-          matches[obj.id] = match;
-          results.push(match);
-        } else {
-          match = matches[obj.id];
-        }
-        match.elements.push(el);
+  var matches = {}, results = [];
+  var els = this.querySelectorAll(this.selectors.join(', '), context);
+
+  var i, j, len, len2, el, m, match, obj;
+  for (i = 0, len = els.length; i < len; i++) {
+    el = els[i];
+    m = this.matches(el);
+    for (j = 0, len2 = m.length; j < len2; j++) {
+      obj = m[j];
+      if (!matches[obj.id]) {
+        match = {
+          id: obj.id,
+          selector: obj.selector,
+          data: obj.data,
+          elements: []
+        };
+        matches[obj.id] = match;
+        results.push(match);
+      } else {
+        match = matches[obj.id];
       }
+      match.elements.push(el);
     }
+  }
 
-    return results.sort(sortById);
-  };
-
-  // Public: Match element against all selectors in set.
-  //
-  // el - An Element
-  //
-  // Returns Array of {selector, data} matches.
-  SelectorSet.prototype.matches = function(el) {
-    if (!el) {
-      return [];
-    }
+  return results.sort(sortById);
+};
+
+// Public: Match element against all selectors in set.
+//
+// el - An Element
+//
+// Returns Array of {selector, data} matches.
+SelectorSet.prototype.matches = function(el) {
+  if (!el) {
+    return [];
+  }
 
-    var i, j, k, len, len2, len3, index, keys, objs, obj, id;
-    var indexes = this.activeIndexes, matchedIds = {}, matches = [];
-
-    for (i = 0, len = indexes.length; i < len; i++) {
-      index = indexes[i];
-      keys = index.element(el);
-      if (keys) {
-        for (j = 0, len2 = keys.length; j < len2; j++) {
-          if (objs = index.map.get(keys[j])) {
-            for (k = 0, len3 = objs.length; k < len3; k++) {
-              obj = objs[k];
-              id = obj.id;
-              if (!matchedIds[id] && this.matchesSelector(el, obj.selector)) {
-                matchedIds[id] = true;
-                matches.push(obj);
-              }
+  var i, j, k, len, len2, len3, index, keys, objs, obj, id;
+  var indexes = this.activeIndexes, matchedIds = {}, matches = [];
+
+  for (i = 0, len = indexes.length; i < len; i++) {
+    index = indexes[i];
+    keys = index.element(el);
+    if (keys) {
+      for (j = 0, len2 = keys.length; j < len2; j++) {
+        if (objs = index.map.get(keys[j])) {
+          for (k = 0, len3 = objs.length; k < len3; k++) {
+            obj = objs[k];
+            id = obj.id;
+            if (!matchedIds[id] && this.matchesSelector(el, obj.selector)) {
+              matchedIds[id] = true;
+              matches.push(obj);
             }
           }
         }
       }
     }
+  }
 
-    return matches.sort(sortById);
-  };
+  return matches.sort(sortById);
+};
 
-  // Public: Expose SelectorSet as a global on window.
-  window.SelectorSet = SelectorSet;
-})(window);
+// Public: Expose SelectorSet as a global on window.
+// BUILD window.SelectorSet = SelectorSet;
```

Comment:
This `// BUILD` marker is replaced with `sed` during `npm run build`. I couldn't find a babel module transformer to do what we needed so I just used Unix. It is unusual, though.